fastlane/Fastfile

```diff
@@ -1,92 +1,74 @@
-def ensure_git_status_clean
-  gitstatus = `xcrun git status --porcelain`
-  if gitstatus != "?? 0001-privateConstants.patch\n"
-    puts("⚠️ There are local changes. Please commit, stash or reset before continuing.\n#{gitstatus}")
-    exit(-2)
-  end
+fastlane_version '2.82.0'
+
+#### Release ####
+
+desc 'Release a new version of VLC to the App Store'
+desc ''
+desc 'This action requires the following parameters:'
+desc '- platform (iOS or tvOS)'
+desc ''
+desc 'This action does the following:'
+desc '- Ensure a clean git status'
+desc '- Increment the version and build number and commit the change (only for iOS releases)'
+desc '- Apply the privateConstants which include the credentials'
+desc '- Build and sign the app'
+desc '- Update the changelog from the NEWS file'
+desc '- Push the version bump'
+lane :release do |options|
+  platform = get_platform options
+  ensure_git_status_clean
+  clear_derived_data
+  bump_version_and_commit if platform == 'iOS'
+  git_apply_private_constants
+  cocoapods(repo_update: true)
+  gym(scheme: "VLC-#{platform}")
+  pilot(app_platform: platform == 'tvOS' ? 'appletvos' : 'ios')
+  update_changelog
+  push_to_git_remote
 end
 
-def git_apply_private_constants
-  Dir.chdir("..") do
-    gitapply = `xcrun git apply 0001-privateConstants.patch`
-    if gitapply != ""
-      puts("⚠️ There are conflicts. Please resolve the conflicts and update the privateConstants.patch before continuing.\n#{gitapply}")
-      exit(-2)
-    end
-  end
-end
+#### Private ####
 
-def changelog 
-    #splits the News by -------- get out the top notes
-    changelog = File.read("../Docs/NEWS").split("-----------")[1].split("-----------").first
-    tempChangelog = changelog.split("iOS")
-    if (tempChangelog.count <= 1)
-      tempChangelog = changelog.split("tvOS")
-    end
-    changelog = tempChangelog[0..-2].join.strip
-    set_changelog(app_identifier: "org.videolan.vlc-ios", changelog: changelog, username: "*", team_name:"VideoLAN")
+desc 'Bump and commit app version and build number'
+private_lane :bump_version_and_commit do
+  increment_version_number(bump_type: 'patch')
+  increment_build_number
+  commit_version_bump(message: 'Version Bump by fastlane', force: true)
 end
 
-def build_app
-    project_file = 'VLC.xcodeproj'
-
-    increment_version_number(
-      bump_type: "patch",
-      xcodeproj: project_file
-    )
-
-    increment_build_number(
-      xcodeproj: project_file)
-
-    commit_version_bump(
-      message: 'Version Bump by fastlane',
-      xcodeproj: project_file,
-      force: true)
+desc 'Update changelog in iTunes Connect with the content from Docs/NEWS'
+private_lane :update_changelog do |options|
+  # Splits the News by -------- get out the top notes
+  changelog = File.read('../Docs/NEWS').split('-----------')[1].split('-----------').first
+  temp_changelog = changelog.split(options[:platform])
+  if temp_changelog.count <= 1
```

Comment:
And this rule is really really hard when moving between Swift and Ruby 😅

By the way... I think the logic here, could be simplified changing the format of docs/NEWS to the format defined by [keepachangelog.com](http://keepachangelog.com)

We could then use the following plugin to set the changelogs:
https://github.com/pajapro/fastlane-plugin-changelog

Anyway, this would be something to face in a separate issue imho.

Reply:
I didn't know there was a guide for changelogs! Very cool. This definitely would be something for a different issue if we want to do this but let me see what @fkuehne thinks about this :)